[WD-25995] fix links and refactor the page to use patterns

akbarkz opened the pull request:
## Done

- Fixed a link for .NET according to [this comment](https://docs.google.com/document/d/1JhVoqA6RyDu3pBzIarrqdY0RgweBLDdHAfeqgAmBDNk/edit?disco=AAABpsZYlZc) from the copy doc
- Changed the link to a copy doc, since the previous one was located in the wrong place on Google Drive, and had an incorrect name as well
- Refactored the whole codebase to utilize Vanilla patterns as much as possible

## QA

- Open the page at https://ubuntu-com-15531.demos.haus/toolchains
- Check that the comments from the [copy doc ](https://docs.google.com/document/d/1JhVoqA6RyDu3pBzIarrqdY0RgweBLDdHAfeqgAmBDNk/edit?tab=t.0)are applied
- Check that the sections look correct

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-25995

## Help

[QA steps](https://discourse.canonical.com/t/qa-steps/152) - [Commit guidelines](https://discourse.canonical.com/t/commit-guidelines/148)

webteam-app commented:
### [<img src='https://assets.ubuntu.com/v1/6baef514-ubuntu-circle-of-friends-large.svg' height=32 width=32> Demo</img>](https://ubuntu-com-15531.demos.haus)
### [<img src='https://assets.ubuntu.com/v1/e512b0e2-jenkins.svg' height=32 width=32> Jenkins </img>](https://jenkins.canonical.com/webteam/job/start-demo/18824)
### [<img src='https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg' height=32 width=32> demos.haus </img>](https://demos.haus)

akbarkz commented:
@mattea-turic Thanks for the review!

> The hero paragraph text has greater line height/vertical space than

This is due to using hero pattern. Do you know how to fix it but still use a pattern?

> This section has been wrapped in two reg. sections – could we just have one pls, as there's excessive bottom padding

I can't use only one section for this since this section of the page does not correspond to our Linked Logo Section pattern, which does not have a description. Please check it again, as I tried to address the excessive padding issue and still use the existing pattern.

> Would it be possible to have the linked logo occupy full page width so we have: first line = 4 logos; second line = 3 logos? The [linked logo section](https://vanillaframework.io/docs/patterns/linked-logo-section) should allow for this now, as H2, etc., is no longer required

This is not relevant anymore as PM asked to not include the new logo for NodeJS. Removing it solves the issue.

> I believe the aria label (logo name) for the openjdk logo should be "OpenJDK" and not "Java" as it currently is

Fixed

> This is super tiny so sorry for bringing it up, but we should be using en-dashes, so the paragraph text should read as "...Ubuntu Pro – Canonical’s comprehensive" rather than "...Ubuntu Pro — Canonical’s comprehensive...". I amended this in the copydoc now too

Fixed

> I think we can safely remove the rule above the CTA

Actually we can't according to the Tiered List pattern docs. I don't believe there is an option to remove it, please let me know if there is. The way the pattern was used in these two sections was a bit wrong, I fixed it as well.

Onibenjo commented:
@akbarkz The decriptions says
> Added another logo for NodeJS according to the comment from Rajan in the same [copy doc](https://docs.google.com/document/d/1JhVoqA6RyDu3pBzIarrqdY0RgweBLDdHAfeqgAmBDNk/edit?tab=t.0) (couldn't provide a link here)

But I see that's not updated in this PR

Onibenjo commented on templates/toolchains/index.html:
```diff
@@ -15,301 +21,303 @@
-      <div class="col">
-        <div class="p-section"
-          <p>From development environments to mission critical production systems, millions of developers trust Ubuntu as the base OS for their software projects. Discover the security, compliance, developer experience and automation features available on Ubuntu.</p>
+    <div class="p-strip is-shallow">
```
```suggestion
    <div class="p-section--shallow">
```

akbarkz replied:
there is already a p-section dev that wraps this strip, so using strip is better in this case.

Onibenjo reviewed (approved):
LGTM.
Left a minor comment
